indigo/analysis/refs/base.py

```diff
@@ -167,7 +167,7 @@ def setup(self, root):
     def is_valid(self, node, match):
         # check that it's not an external reference
         ref = match.group(0)
-        if ref.endswith('the ') or ref.endswith('Act '):
+        if ref.endswith('of ') or ref.endswith('thereof'):
```

Comment:
What if it ends with `of this`? Instead, perhaps check if the very last `match.group()` is non-empty? We're basically re-doing the regex with `endswith` otherwise.

Reply:
There were existing tests for `of this` which are passing; I've reshuffled the tests so that hopefully their grouping makes a bit more sense.
I'm not sure I follow you on the last `match.group()` being non-empty – it _will_ be empty for simple references like `section 26`, but will contain `this` in the case of `section 26 of this XXX`.

Reply:
Ah, I understand now. "of " is the exception, not "of this". All good.